integration/integration_test.go

```diff
@@ -463,6 +465,331 @@ func Test_Integration_PublishThrowsNoError(t *testing.T) {
+					  "value": 120
+					},
+					{
+					  "timestamp": 10000000,
+					  "name": "cumulative-rate",
+					  "type": "cumulative-rate",
+					  "attributes": {},
+					  "value": 120
+					},
+					{
+					  "timestamp": 10000000,
+					  "name": "prometheus-histogram",
+					  "type": "prometheus-histogram",
+					  "attributes": {},
+					  "value": {
+						"sample_count": 2,
+						"sample_sum": 3,
+						"buckets": [
+						  {
+							"cumulative_count": 1,
+							"upper_bound": 1
+						  },
+						  {
+							"cumulative_count": 2,
+							"upper_bound": 2
+						  }
+						]
+					  }
+					},
+					{
+					  "timestamp": 10000000,
+					  "name": "prometheus-summary",
+					  "type": "prometheus-summary",
+					  "attributes": {},
+					  "value": {
+						"sample_count": 2,
+						"sample_sum": 2,
+						"quantiles": [
+						  {
+							"quantile": 0.5,
+							"value": 1
+						  },
+						  {
+							"quantile": 0.9,
+							"value": 1
+						  }
+						]
+					  }
+					}
+				  ],
+				  "inventory": {
+					"some-inventory": {
+					  "some-field": "some-value"
+					}
+				  },
+				  "events": [
+					{
+					  "timestamp": 10000000,
+					  "summary": "evnt2sum",
+					  "category": "evnt2cat"
+					}
+				  ]
+				}
+			  ]
+			}`)
+
+			// awful but cannot compare with json.Unmarshal as it's not supported by Integration
+			assert.Equal(t, stripBlanks(expectedOutputRaw), stripBlanks(integrationBytes))
+		},
+	}
+
+	_ = os.Setenv("HOST_ID", "id-1234567890abc")
+
+	// os.ClearEnv breaks tests in Windows that use the fileStorer because clears the user env vars
+	defer func() {
+		_ = os.Unsetenv("HOST_ID")
+	}()
+
+	i, err := New("TestIntegration", "1.0", Logger(log.Discard), Writer(w), Args(&arguments))
+	assert.NoError(t, err)
+
+	e, err := i.NewEntity("localhost:8080", "test", "")
+	assert.NoError(t, err)
+	_ = e.AddTag("env", "prod")
+
+	gauge, _ := Gauge(time.Unix(10000000, 0), "metric-gauge", 1)
+	count, _ := Count(time.Unix(10000000, 0), "metric-count", 100)
+	_ = count.AddDimension("cpu", "amd")
+	summary, _ := Summary(time.Unix(10000000, 0), "metric-summary", 1, 10, 100, 1, 100)
+	// attributes should be ordered by key in lexicographic order
+	_ = summary.AddDimension("os", "linux")
+	_ = summary.AddDimension("distribution", "debian")
+	// add metrics to entity 1
+	e.AddMetric(gauge)
+	e.AddMetric(count)
+	e.AddMetric(summary)
+	// add 1st event to entity 1
+	ev1, err := event.New(time.Unix(10000000, 0), "evnt1sum", "evnt1cat")
+	assert.NoError(t, err)
+	_ = ev1.AddAttribute("attr1", "attr1Val")
+	_ = ev1.AddAttribute("attr2", 42)
+	e.AddEvent(ev1)
+	// add 2nd event to entity 1
+	ev2, err := event.New(time.Unix(10000000, 0), "evnt2sum", "evnt2cat")
+	assert.NoError(t, err)
+	e.AddEvent(ev2)
+	// add inventory to entity 1. only one because order is not guaranteed and the test is comparing with a static string
+	err = e.AddInventoryItem("custom/example", "version", "1.2.3")
+	assert.NoError(t, err)
```

Comment:
I would remove all this from the test and scope it just to test the `HostId` replacement logic. I think these should be checked in different unit tests.

Reply:
You are right, I will clean up this test a bit 👍